core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/CryptoUtil.java

```diff
@@ -431,6 +546,20 @@ public byte[] base64DecodeAndDecrypt(String base64CipherText) throws
         return decrypt(Base64.decode(base64CipherText));
     }
 
+    /**
+     * Base64 decode the given value and decrypt using the custom param.
+     *
+     * @param base64CipherText  Base64 encoded cipher text.
+     * @param secretKey         Secret key.
+     * @return Base64 decoded, decrypted bytes.
+     * @throws CryptoException On an error during decryption.
+     */
+    public byte[] base64DecodeAndDecryptWithCustomKey(String base64CipherText, String secretKey) throws
```

Comment:
Comments added to  `encryptWithCustomKeyAndBase64Encode` is applicable for this as well

Reply:
`base64DecodeAndDecrypt` is already available with the params String, String